ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -77,6 +110,49 @@ objects:
         secretName: ${JOB_NAME_SAFE}-cluster-profile
 
     containers:
+    - name: egress-proxy
+      command:
+      - /bin/bash
+      - -c
+      args:
+      - |
+        #!/bin/bash
+        set -euo pipefail
+
+        if [[ "${ENABLE_PROXY}" == "true" ]]; then
+          trap 'touch /tmp/shared/exit' EXIT
+        fi
+
+        # this is just a stop gap -- we want to have a secret created to house this
+        echo -e 'admin:$apr1$JOHdRB4Z$S7zUn2dtdxlXLSgWdlFp40' > /tmp/passwords
+
+        cat > /tmp/squid.conf <<EOF
+        http_port 3128
+        cache deny all
+        access_log none all
+        debug_options ALL,0
+        shutdown_lifetime 0
+        auth_param basic program /usr/lib64/squid/basic_ncsa_auth /tmp/passwords
+        auth_param basic realm proxy
+        acl authenticated proxy_auth REQUIRED
+        http_access allow authenticated
+        pid_filename /tmp/shared/proxy-setup
+        EOF
+
+        if [[ "${ENABLE_PROXY}" == "true" ]]; then
+          exec squid -N -f /tmp/squid.conf
```

Comment:
Instead of `exec`-ing Squid, we probably want to run it in the background and `wait` on it.  That way, we can monitor for test completion, gently shut Squid down after the tests complete, and then copy the logs over into `/tmp/shared` for archival.  Something like:

```sh
trap 'rc=$?; if test "${rc}" -ne 0; then touch /tmp/shared/exit; fi; exit "${rc}"' EXIT
trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
```

at the top of this script, and then:

```sh
squid -N -f /tmp/squid.conf &
SQUID_PID="$!"

while [[ ! -f /tmp/shared/exit ]]; then
  sleep 60
done

kill "${SQUID_PID}"
wait
cp wherever/squid/keeps/logs /tmp/shared/squid
```